test/Cardano/FeeSpec.hs

```diff
@@ -546,6 +563,72 @@ propDistributeFeeNoNullFee (fee, outs) =
   where
     prop = property $ Fee 0 `F.notElem` (fst <$> distributeFee fee outs)
 
+{-------------------------------------------------------------------------------
+                         coalesceDust - Properties
+-------------------------------------------------------------------------------}
+
+data CoalesceDustData = CoalesceDustData
+    { cddThreshold :: DustThreshold
+    , cddCoins :: NonEmpty Coin
+    } deriving (Eq, Generic, Show)
+
+instance Arbitrary CoalesceDustData where
+    arbitrary = do
+        coinCount <- genCoinCount
+        coins <- (:|) <$> genCoin <*> replicateM coinCount genCoin
+        threshold <- DustThreshold . getCoin <$> oneof
+            [ -- Two possibilities:
+              genCoin
+              -- ^ A completely fresh coin.
+            , elements (F.toList coins)
+              -- ^ A coin picked from the existing coin set.
+            ]
+        pure $ CoalesceDustData threshold coins
+      where
+        genCoin = Coin <$> oneof [pure 0, choose (1, 100)]
+        genCoinCount = choose (0, 10)
+    shrink = genericShrink
+
+propCoalesceDustPreservesSum :: CoalesceDustData -> Property
+propCoalesceDustPreservesSum (CoalesceDustData threshold coins) =
+    property $
+    let total = sum (getCoin <$> coins) in
+    cover 8 (total == 0) "sum coins = 0" $
+    cover 8 (total /= 0) "sum coins ≠ 0" $
+    total == F.sum (getCoin <$> coalesceDust threshold coins)
+
+propCoalesceDustLeavesNoZeroCoins :: CoalesceDustData -> Property
+propCoalesceDustLeavesNoZeroCoins (CoalesceDustData threshold coins) =
+    property $
+    cover 4 (F.all  (== Coin 0) coins) "∀ coin ∈ coins . coin = 0" $
+    cover 4 (F.elem    (Coin 0) coins) "∃ coin ∈ coins . coin = 0" $
+    cover 8 (F.notElem (Coin 0) coins) "∀ coin ∈ coins . coin > 0" $
+    F.notElem (Coin 0) $ coalesceDust threshold coins
+
+propCoalesceDustLeavesAtMostOneDustCoin :: CoalesceDustData -> Property
```

Comment:
how about checking situation when we all coins are below threshold?

Reply:
Fixed: https://github.com/input-output-hk/cardano-coin-selection/pull/27/commits/b7bf2b6cb60c23f2c17b153c112ed99367fcb511